Libraries/LibWeb/Layout/LayoutState.cpp

```diff
@@ -373,17 +374,37 @@ void LayoutState::commit(Box& root)
             continue;
         }
 
-        auto const& fragments = paintable_with_lines->fragments();
-        if (fragments.is_empty()) {
-            continue;
-        }
+        Optional<CSSPixelPoint> offset;
+        paintable_with_lines->for_each_in_inclusive_subtree_of_type<Painting::PaintableWithLines>([&offset](auto& paintable) {
+            auto const& fragments = paintable.fragments();
+            if (!fragments.is_empty()) {
+                if (!offset.has_value() || (fragments.first().offset().x() < offset.value().x())) {
+                    offset = fragments.first().offset();
+                }
+            }
+            return TraversalDecision::Continue;
+        });
+        if (offset.has_value())
+            paintable_with_lines->set_offset(offset.value());
 
-        paintable_with_lines->set_offset(fragments.first().offset());
         CSSPixelSize size;
-        for (auto const& fragment : fragments) {
-            size.set_width(size.width() + fragment.width());
-            size.set_height(max(size.height(), fragment.height()));
-        }
+        paintable_with_lines->for_each_in_inclusive_subtree_of_type<Painting::PaintableWithLines>([&size](auto& paintable) {
```

Comment:
I think the first two loops here can be combined, not sure about the third since it skips some items.

Reply:
Yeah you are right, merging that is better both computationaly and reduced the boilerplate-to-actual-work ratio of the code as well.

The height loop was a bit heuristic. If the paintable does not have any fragments to derive it's height from, I just took the height of one of the children instead. Note that always taking the max over all children is wrong, as embedding a heavier *bold* word should not affect the height of boxes around the other text. However I realised that it is more correct or simply ask the layout-node for the line-height if we have no fragments instead as the height for the paintable.